File: src/main/java/com/levelupquest/entities/Transaction.java
package com.levelupquest.entities;

import java.io.Serializable;
import java.util.ArrayList;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor 
@Entity
public class Transaction implements Serializable {

	@Id
	@GeneratedValue
	private int id;
	private String accountId;
	private ArrayList<String> categoryTags;
	private long currencyAmount;
	private String customerId;
	private String description;
	private String locationCity;
	private String locationCountry;
	private double locationLatitude;
	private double locationLongitude;
	private String locationPostalCode;
	private String locationRegion;
	private String locationStreet;
	private String merchantCategoryCode;
	private String merchantId;
	private String merchantName;
	private long originalCurencyAmount;
	private String originalCurrency;
	private String originationDate;
	private long postBalance;
	private String postDate;
	private String source;
	private String type;
	private boolean success;
	private String message;

}

File: src/main/webapp/frontend/src/app/model/transaction.ts
export class Transaction {
  public  accountId: String;
  public categoryTags: String[] ;
  public currencyAmount: number ;
  public customerId: String ;
  public description: String ;
  public id: string ;
  public locationCity: string;
  public locationCountry: string;
  public locationLatitude: number;
  public locationLongitude: number;
  public locationPostalCode: string;
  public locationRegion: string;
  public locationStreet: string;
  public merchantCategoryCode: string;
  public merchantId: string;
  public merchantName: string;
  public originalCurencyAmount: number;
  public originalCurrency: string;
  public originationDate: string;
  public postBalance: number;
  public postDate: string;
  public source: string;
  public type: string;

  constructor(accountId: String, customerId: String, categoryTags: String[], description: String, currencyAmount: number) {
      this.accountId = accountId;
      this.customerId = customerId;
      this.categoryTags = categoryTags;
      this.description = description;
      this.currencyAmount = currencyAmount;


  }
}
